shared-module/displayio/OnDiskBitmap.c

```diff
@@ -141,38 +200,87 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
         uint8_t red;
         uint8_t green;
         uint8_t blue;
-        if (self->bits_per_pixel == 1) {
+
+        if (self->dither) {
+            uint8_t noise =  noise2d(x,y) * 255;
+            uint8_t randr  = noise & self->dither_mask_r;
+            uint8_t randg  = (noise >> 3) & self->dither_mask_g;
```

Comment:
It seems like the shifts and masks must be related somehow, but yet the masks are hard-coded.